### PR TITLE
server: check guest os preference of last host when start a vm

### PR DESCRIPTION
**Pull request by ustcweizhou**
## Description

If vm has last host_id specified, cloudstack will try to start vm on it at first.
However, host tag is checked, but guest os preference is not checked.

for new vm, it will be deployed to the preferred host as we expect.

Fixes: https://github.com/apache/cloudstack/issues/3554#issuecomment-564001531

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

File: server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java [hunk 1]
```diff
@@ -489,8 +487,6 @@ public DeployDestination planDeployment(VirtualMachineProfile vmProfile, Deploym
                         } else {
                             s_logger.debug("The last host of this VM does not have enough capacity");
                         }
-                    } else {
```
Comment (ravening) [inline server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java:492]: Any reason to delete this log? Better to display if either tags or the guest os preference mismatches right?

Comment (weizhouapache) [reply server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java:492]: @ravening it is moved to method checkVmProfileAndHost

**Review by DaanHoogland — APPROVED** [review]
code lgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by ravening** [comment]
LGTM

**Review by rhtyd — APPROVED** [review]

**Pull request merged**